src/BenchmarksApps/TodosApi/ValidationFilter.cs

```diff
@@ -0,0 +1,166 @@
+    {
+        var validatableParametersMap = context.HttpContext.GetEndpoint()?.Metadata.OfType<ValidationMetadata>()
+            .FirstOrDefault()?.ValidatableParametersMap;
+
+        if (validatableParametersMap is null)
+        {
+            return next(context);
+        }
+
+        Dictionary<string, List<string>>? workingErrors = null;
+
+        foreach (var index in validatableParametersMap)
+        {
+            var validatable = context.GetArgument<IValidatableObject?>(index);
+            if (validatable is not null)
+            {
+                var requiresServices = validatable is IValidatable v && v.RequiresServiceProvider;
+                var validationContext = GetValidationContext(validatable, requiresServices ? context.HttpContext.RequestServices : null);
+                if (_logger.IsEnabled(LogLevel.Information))
+                {
+                    _logger.LogInformation("Validating argument at index {index} of type {type}", index, validatable.GetType().Name);
+                }
+                var results = validatable.Validate(validationContext);
+
+                if (results.Any())
+                {
+                    if (_logger.IsEnabled(LogLevel.Information))
+                    {
+                        _logger.LogInformation("Argument at index {index} of type {type} has validation errors", index, validatable.GetType().Name);
+                    }
+                    workingErrors ??= new();
+
+                    foreach (var result in results)
+                    {
+                        foreach (var member in result.MemberNames)
+                        {
+                            List<string>? messages;
+                            if (!workingErrors.ContainsKey(member))
+                            {
+                                messages = new();
+                                workingErrors.Add(member, messages);
+                            }
+                            else
+                            {
+                                messages = workingErrors[member];
+                            }
+                            messages.Add(result.ErrorMessage ?? "The value provided was invalid.");
+                        }
+                    }
+
+                    var errors = MapToFinalErrorsResult(workingErrors);
+                    var problem = TypedResults.ValidationProblem(errors);
+                    return ValueTask.FromResult((object?)problem);
+                }
+            }
+        }
+        return next(context);
+    }
+
+    [UnconditionalSuppressMessage("Trimming", "IL2026:Members annotated with 'RequiresUnreferencedCodeAttribute' require dynamic access otherwise can break functionality when trimming application code",
+        Justification = "Instance type is statically represented in generic argument declared as dynamically accessing public properties.")]
```

Comment:
We should note here (because I'm sure people will look at this code and assume this suppression is always valid) that only the properties on `TTarget` are being preserved by the trimmer. Once you recurse and inspect properties of those property types, you are broken again.

Reply:
There's no recursion going on here though, and System.CM.DA doesn't recurse today. This isn't library code, it's an internal type in an app, so I'm not quite sure I fully understand your threshold for how suppressions should be used yet.

Reply:
It's more about the wording of the `Justification`.  I think suppressing for this app is fine.

Reply:
```suggestion
        Justification = "Instance type is statically represented in generic argument declared as dynamically accessing public properties. This ensures the properties on TTarget are preserved. Note that recursive properties of those properties are still not preserved, but this validation doesn't recurse.")]
```

Reply:
I updated the justification to be more descriptive. LMK if you think it needs more.